### DISCUSSION TITLE
PCB für OpenDTU

### DISCUSSION THREAD

**Discussion by micw** [tbnobody/OpenDTU#651] [Show and tell]
Hallo zusammen,

ich möchte gern mein PCB teilen, welches ich heute entworfen habe:
https://oshwlab.com/mwyraz/opendtu

Wenn man direkt aus EasyEDA bestellt, bekommt man einen 8$ Gutschein, so dass 10 Stück ncl. Versand knapp 5€ kosten.

Test des PCBs steht noch aus, ich berichte in ca. 1 Woche davon ;-)

Viele Grüße und vielen Dank für das tolle Projekt,
Michael.

**Comment by Forhorse88** [comment]
no capacitors ? (for filtering and decoupling)

**Comment by micw** [comment]
It's just using an ESP32 mini dev board (https://www.azdelivery.de/products/esp32-d1-mini) that comes with everything on board plus the RF module and an (optional) dc/dc converter. All required capacitors are on the dev board. Scheme is basically the same as in the readme.

(in fact another variant of the breakout board linked in the project)

**Reply by Forhorse88** [reply]
A simple rule in electronics : you never have enough capacitors !
And power the NRF24 module with the on-board 3.3V regulator of the ESP32 I wouldn't try it (Especially without capacitor)
yes it will probably work, but what about the stability ?
You should use a 3.3V DC/DC converter instead of a 5V and supplying the ESP32 and the NRF25 directly in 3.3V (with some filtering capacitors).

**Reply by micw** [reply]
I do agree with the 3v3 regulator (since those are also available with the same form factor and the 5v->3v DCDC on the esp dev boards are usually quite limited).
But I still do not see the need to extra capacitors since both modules as well as the power supply module have their own. I made some other "breakout pcbs" where capacitors were indeed critical (e.g. a driver for adressable LEDs where I used a level shifter (not a module) so no caps where added.

I'm running the setup for a while on my desk (with prototype board) and it appears very stable.

**Reply by jaybee79** [reply]
A 3,3V buck converter might do the job more efficiant but I power my OpenDTU via USB of the ESP32 dev board and the onboard linear converter and it's running stable. A SH1106 (I2C) display is also added. I fully agree with "you never have enough capacitors". My NRF24 got some small SMD capacitors (22µ/1µ/100n) directly soldered on it's connector pins, the ESP32 a 22µ capacitor on it's connection on the breadboard and just in case I added a 2200µ/10V (yes, 2,2mF) capacitor to the 3,3V line that was laying around. :-D The system is drawing about 90mA according to my USB meter, sometimes I see spikes to 110mA.

**Reply by micw** [reply]
See below about a very large cap on the 3V3 line. If you know it's resistance, you can calculate, how long it takes to charge the 0,012 joule (https://www.allaboutcircuits.com/tools/capacitor-Charge-and-time-constant-calculator/). You easily end up in a consumption of several amperes for some milliseconds.

**Reply by jaybee79** [reply]
> See below about a very large cap on the 3V3 line. If you know it's resistance, you can calculate, how long it takes to charge the 0,012 joule (https://www.allaboutcircuits.com/tools/capacitor-Charge-and-time-constant-calculator/). You easily end up in a consumption of several amperes for some milliseconds.

I'm fully aware of the behavior of capacitors. The current is limited by several things: inductivity and resistance of the connection between capacitor and voltage source and so on. The main limitation of the current will come from the voltage regulator on the ESP32 board, the USB cable and the powersupply or the USB connection on the computer it is plugged into. In my case it works for several weeks and I have switched off/on the power very often during some tests with some RF modules. I bought some more on ebay to build some OpenDTUs for my relatives.

**Comment by huckes** [comment]
Hi, nice PCB project. Especially I like the usage of the D1 form factor as this allows to design smaller cases.

How do you think about adding soldering points for OLED-Displays using the I2C interface and probably for the ePaper display which uses the SPI-Interface?
And along with these enhancements you might also like to spend two soldering points for an additional capacitor; for those who want to add one.

**Reply by micw** [reply]
I can easily add it. Do you have a link to schematics? What kind of connector would be used?
What is the voltage and estimated power consumption of these displays? If they use 5V, also a level shifter might be needed.

**Reply by huckes** [reply]
I just recognized, that you used  other GPIOs for the SPI-interface as I assumed to use the manufactor's suggested ports for SPI  as you find in https://arduino-projekte.info/produkt/d1-mini-esp32-wifi-bluetooth/. So is there a reason why you used another pin-mapping?

Currently two I2C-Displays are supported by openDTU: SSH1106 and SSD1306. Both work with 5V and 3.3V. But in the case of the ESP-Microcontrollers they should be supplied with 3.3V. Both types have four connections: VDD(3.3V), GND, SDA and SCL. For the ESP32 D1 mini the suggested pins are GPIO22 for  SCL and GPIO21 for SDA. Current consumption is documented with about 11mA. I run two openDTUs with SSD1306-Dispalys without problems using the VCC-regulator mounted on the D1-Board. So from my perspective no extra voltage supply or level shifter is needed, but a capacitor might be sensefull. 
As the displays you find in the shops have slightly different connections I like to suggest to provide simply four soldering points (Lötaugen) in a row of 2.54mm distance (VCC,GND, SCL, SDA). The display can be connected by wires accordingly to the connectors they have and independly of there physical position int the case. 
The display supported by the SPI-interfacew seems to be still under evaluation, so the wiring might be defined later. But I assume they use the same SPI connections as the NRF-Module, but another GPIO for CS.

**Reply by micw** [reply]
I had started with a test board I initially made for esp8266 (wemos D1), so I used the "inner" pin row only. It should not matter much because most GPIO on the esp32 are really general-purpose.

**Reply by huckes** [reply]
Yes, all the default pins can be bit-banged to other pins, in the case of openDTU with the pin-mapping json file. Perhaps it might be easier for the PCB-layout to use other pins than the default ones. In my test-setup I used the default ones, as I wired it with dupont cables. At the end it's only relevant that it works ;-)

**Comment by micw** [comment]
Where would you add caps and for which exact purpose?

I recently created a PCB for my kid's WLED projects: https://oshwlab.com/mwyraz/wled-8-ch

Here I power everything with 5V since the level shifter needs it. The board has an option to pass the 5V directly to the ESP-Mini or to use an external 5v->3V3 converter. I made this decision because I had some ESP32 mini clones with poor onboard converters that ran unstable, especially when there was noise on the 5V line.
Here I added 2 caps on purpose: a larger one to buffer the input power and a small (and fast) ceramic cap near the level shifter that catches fast spikes.

Do you see a similar situation with the NRF24? I'm running mine on the (same poor) onboard 5V->3V3 converter on my ESP and all runs very stable. Did anyone had issues with stability with this setup?

Don't get me wrong, I'm not against adding caps where necessary. But adding too much caps may also cause harm. E.g. you can add a large cap on the 3V3 output of the ESP module. That _might_ increase stability but it even can prevent the ESP from booting (since the cap draws a lot of power for a few µs that the onboard DC/DC may not be able to provide.
